libs/remix-lib/src/execution/txRunnerWeb3.ts

```diff
@@ -203,8 +203,12 @@ export class TxRunnerWeb3 {
             callback(new Error('Gas estimation failed because of an unknown internal error. This may indicated that the transaction will fail.'))
             return
           }
-          if (tx.fromSmartAccount && tx.value === "0" && err && err.error && err.error.indexOf('insufficient funds for transfer') !== -1) {
-            // Do not show dialog for insufficient funds as smart account may be using paymaster
+          if (tx.fromSmartAccount && tx.value === "0" && (
```

Comment:
I think we need to find a better way to check that...
below you are checking against `err.error` and not `err`..
should that be: `err && err.error..indexOf('gas required exceeds allowance (0)')` ?

Reply:
updated